src/sentry/api/endpoints/project_user_reports.py

```diff
@@ -50,9 +50,18 @@ def get(self, request, project):
         :pparam string project_slug: the slug of the project.
         :auth: required
         """
+        try:
+            environment = self._get_environment_from_request(
+                request,
+                project.organization_id,
+            )
+        except Environment.DoesNotExist:
+            return Response({'environment': 'Invalid environment'}, status=400)
```

Comment:
I think we should `raise ResourceDoesNotExist` instead, as this is what other endpoints are doing.

Reply:
I think there are two good approaches (I don't have a strong opinion one way or another, and I think we've done both for environments):

1. Return an empty result set from the endpoint (since the environment doesn't exist, no user reports can exist.)
2. Raise a `ResourceDoesNotExist` exception like @mattrobenolt mentioned to cause a 404 response.

From a idealistic standpoint I think that the first option might be better: it seems questionable to return a 404 response here since the resource /was/ found, but the filter criteria were invalid. (From that perspective, I think you could make an argument for a 400 as well.)

¯\\\_(ツ)_/¯

Reply:
@mattrobenolt and @tkaemming The reason I raise a 400 is because a bit lower in the get method if the group status is invalid it raises a 400. I was thinking that environment should do what group status does. Do you disagree?